### DISCUSSION TITLE
how to load conda asynchronously?

### DISCUSSION THREAD

**Discussion by fishiu** [zdharma-continuum/zinit#509] [Q&A] [answered]
Conda is initialized by such code, which takes a huge time when starting zsh. Is there any way to asynchronously run this code in zsh? Many thanks!

```
# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/opt/homebrew/Caskroom/miniconda/base/bin/conda' 'shell.zsh' 'hook' 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/opt/homebrew/Caskroom/miniconda/base/etc/profile.d/conda.sh" ]; then
        . "/opt/homebrew/Caskroom/miniconda/base/etc/profile.d/conda.sh"
    else
        export PATH="/opt/homebrew/Caskroom/miniconda/base/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda initialize <<<
```

**Comment by pschmitt** [comment, marked as answer]
You can do something along the lines of:

```zsh
zinit wait lucid as'null' id-as'conda' \
  atclone'CONDA_INIT_CODE > conda.zsh' \
  src'conda.zsh' \
  for zdharma-continuum/null
```

Remember to update `CONDA_INIT_CODE` with the contents of your init function.

**Reply by fishiu** [reply]
Thanks for your answer! It works perfect!!
BTW I don't know how to compress such long command into CONDA_INIT_CODE, so I manually copied the code into conda.zsh. It works anyway.

**Reply by pschmitt** [reply]
You can also create a separate function in your zshrc that you explicitly call with `atload"myfunc"` in your plugin.

**Reply by fishiu** [reply]
Thanks! This is much more useful!
